src/coreclr/src/vm/eventpipesession.h

```diff
@@ -190,7 +196,7 @@ class EventPipeSession
 
     bool WriteAllBuffersToFile(bool *pEventsWritten);
 
-    bool WriteEventBuffered(
+    bool WriteEvent(
```

Comment:
Can we add a comment here that describes when this function is synchronous (under new profiler session type) and when it is asynchronous (all other times)?  The asynchronous nature of this method is important to the performance of the eventing pipeline and without `Buffered` in its name, it may not be immediately clear what the semantics are.